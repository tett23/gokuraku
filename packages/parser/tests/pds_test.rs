extern crate parser;

use insta::{assert_debug_snapshot, glob};
use std::fs;

#[test]
fn pds_test() {
    glob!("../fixtures/pds", "*.pds", |path| {
        let input = fs::read_to_string(path).unwrap();
        let ast = parser::prose_down_script_parse(&input).unwrap();

        assert_debug_snapshot!(ast);
    });
}
