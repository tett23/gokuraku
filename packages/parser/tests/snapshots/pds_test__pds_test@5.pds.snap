---
source: packages/parser/tests/pds_test.rs
expression: ast
input_file: packages/parser/fixtures/pds/5.pds
---
Module {
    statements: [
        LineComment(
            LineComment(
                "lineComment(0, 40)",
            ),
        ),
        Assign(
            Assign {
                ident: Ident(
                    "main",
                ),
                args: AssignArgs {
                    patterns: [],
                },
                expr: Literal(
                    Text(
                        "Hello world",
                    ),
                ),
                where_clause: Module {
                    statements: [],
                },
            },
        ),
    ],
}
